mysys/my_default.c

```diff
@@ -631,7 +631,11 @@ int my_load_defaults(const char *conf_file, const char **groups,
       if (!my_getopt_is_args_separator((*argv)[i])) /* skip arguments separator */
         printf("%s ", (*argv)[i]);
     puts("");
+#ifdef EMBEDDED_LIBRARY
+    DBUG_RETURN(2);
+#else
     exit(0);
+#endif
```

Comment:
This means: if embedded was enabled `DBUG_RETURN(2)` from `libmysqld.so`, `mysqld` and all client binaries, otherwise do `exit(0)`.

This effectively means most binaries with `--help` would `exit(1)` when `-DWITH_EMBEDDED_SERVER=ON`, otherwise they would `exit(0)`. Some binaries like `aria_chk` won't even exit. I doubt that is the intention.